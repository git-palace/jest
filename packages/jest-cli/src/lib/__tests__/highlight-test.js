/**
* Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
*
* This source code is licensed under the BSD-style license found in the
* LICENSE file in the root directory of this source tree. An additional grant
* of patent rights can be found in the PATENTS file in the same directory.
*
* @emails oncall+jsinfra
*/

'use strict';

const {dim, reset} = require('chalk');
const highlight = require('../highlight');

it('Highlight only the matching part and dims the rest', () => {
  let highlighted = `${dim('pa')}${reset('th/to/tes')}${dim('t')}`;
  expect(highlight('path/to/test', 't.*s') === highlighted).toBeTruthy();

  highlighted = reset(`path/to/test`);
  expect(highlight('path/to/test', 'path/to/test') === highlighted)
    .toBeTruthy();
});

it('Returns the a dimmed string when nothing matches', () => {
  expect(highlight('path/to/test', 'other'))
    .toEqual(dim('path/to/test'));
});
