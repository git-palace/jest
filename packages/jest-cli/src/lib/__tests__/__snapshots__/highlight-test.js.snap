exports[`test dims everything when there is no match 1`] = `"[2mjest-cli/__tests__/watch-test.js[22m"`;

exports[`test dims everything when there is no match 2`] = `"[2m...t-cli/__tests__/watch-test.js[22m"`;

exports[`test dims everything when there is no match 3`] = `"[2m.../__tests__/watch-test.js[22m"`;

exports[`test dims everything when there is no match 4`] = `"[2m...sts__/watch-test.js[22m"`;

exports[`test dims everything when there is no match 5`] = `"[2m...watch-test.js[22m"`;

exports[`test dims everything when there is no match 6`] = `"[2m...-test.js[22m"`;

exports[`test dims everything when there is no match 7`] = `"[2m....js[22m"`;

exports[`test highlight the trimmed part when there is only a rootDir match 1`] = `"[2mjest-cli/__tests__/watch-test.js[22m"`;

exports[`test highlight the trimmed part when there is only a rootDir match 2`] = `"[0m...[0m[2mt-cli/__tests__/watch-test.js[22m"`;

exports[`test highlight the trimmed part when there is only a rootDir match 3`] = `"[0m...[0m[2m/__tests__/watch-test.js[22m"`;

exports[`test highlight the trimmed part when there is only a rootDir match 4`] = `"[0m...[0m[2msts__/watch-test.js[22m"`;

exports[`test highlight the trimmed part when there is only a rootDir match 5`] = `"[0m...[0m[2mwatch-test.js[22m"`;

exports[`test highlight the trimmed part when there is only a rootDir match 6`] = `"[0m...[0m[2m-test.js[22m"`;

exports[`test highlight the trimmed part when there is only a rootDir match 7`] = `"[0m...[0m[2m.js[22m"`;

exports[`test highlights everything when there is a full match 1`] = `"[0mjest-cli/__tests__/watch-test.js[0m"`;

exports[`test highlights everything when there is a full match 2`] = `"[0m...t-cli/__tests__/watch-test.js[0m"`;

exports[`test highlights everything when there is a full match 3`] = `"[0m.../__tests__/watch-test.js[0m"`;

exports[`test highlights everything when there is a full match 4`] = `"[0m...sts__/watch-test.js[0m"`;

exports[`test highlights everything when there is a full match 5`] = `"[0m...watch-test.js[0m"`;

exports[`test highlights everything when there is a full match 6`] = `"[0m...-test.js[0m"`;

exports[`test highlights everything when there is a full match 7`] = `"[0m....js[0m"`;

exports[`test highlights the trimmed part there is a non visible match 1`] = `"[0m...t-cli[0m[2m/__tests__/watch-test.js[22m"`;

exports[`test highlights the trimmed part there is a non visible match 2`] = `"[0m...[0m[2m/__tests__/watch-test.js[22m"`;

exports[`test highlights the trimmed part there is a non visible match 3`] = `"[0m...[0m[2msts__/watch-test.js[22m"`;

exports[`test highlights the trimmed part there is a non visible match 4`] = `"[0m...[0m[2mwatch-test.js[22m"`;

exports[`test highlights the trimmed part there is a non visible match 5`] = `"[0m...[0m[2m-test.js[22m"`;

exports[`test highlights the trimmed part there is a non visible match 6`] = `"[0m...[0m[2m.js[22m"`;
